archive/imo/imo2001_q6.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2021 Sara Díaz Real. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sara Díaz Real
+-/
+import data.int.basic
+import tactic.linarith
+import tactic.ring
+import algebra.associated
+
+/-!
+# IMO 2001 Q6
+Let $a$, $b$, $c$, $d$ be integers with $a > b > c > d > 0$. Suppose that
+
+$$
+a*c + b*d = (b*d + a*c) * (b*d + a*c).
+$$
+
+Prove that $a*b + c*d$ is not prime.
+-/
+
+/-!
+The main idea of this formalization is using the negation
+introduction, that is, we are going to suppose that the number
+$a*b+c*d$  is prime.
+
+However, firstly we have to prove some auxiliary results that
+will be used in the final theorem.
+-/
+
+
+
+variables {a b c d : ℤ}
+
+
+/-!
+## Intermediate lemmas
+-/
+
+lemma equivalent_sums
+  (h : a*c + b*d = (a + b - c + d) * (-a + b + c + d)) :
+  b^2 + b*d + d^2 = a^2 - a*c + c^2 :=
+begin
+  have h1 : (a + b - c + d) * (-a + b + c + d) =
+           -a^2  + b^2  + a*c - c^2 + b*d + d^2 + a*c + b*d,
+    by ring,
+  by nlinarith,
+end
+
+lemma equivalent_products
+  (h : a*c + b*d = (a + b - c + d) * (-a + b + c + d))
+  : (a*c + b*d) * (b^2 + b*d + d^2) = (a*b + c*d) * (a*d + b*c) :=
```

Comment:
Here and below, the style guide says that `:`s should go at the end of a line, rather than the beginning.